### PR TITLE
MG-193 - Update Kubernetes deployment docs

### LINKED ISSUE #193: Update Kubernetes deployment docs
### Is your feature request related to a problem? Please describe.

Docs will be out of date as when we merge https://github.com/absmach/devops/pull/127.

### Describe the feature you are requesting, as well as the possible use case(s) for it.

N/A

### Indicate the importance of this feature to you.

Must-have

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by JeffMboya**
Pull request title should be `MG-XXX - description` or `NOISSUE - description` where XXX is ID of issue that this PR relate to.
Please review the [CONTRIBUTING.md](./CONTRIBUTING.md) file for detailed contributing guidelines.

### What does this do?
This PR updates Kubernetes.md to conform to the latest helm charts

### Which issue(s) does this PR fix/relate to

-Resolves #193 

### List any changes that modify/break current functionality
None

### Have you included tests for your changes?
No

### Did you document any new/modified functionality?
Yes

### CODE REVIEW

File: docs/kubernetes.md [hunk 1]
```diff
@@ -145,9 +197,13 @@ The following table lists the configurable parameters and their default values.
 | twins.httpPort                     | Twins service HTTP port                                                                         | 9021             |
```
Comment (arvindh123) [inline docs/kubernetes.md]: @JeffMboya  Values.yaml  are changed a lot, please update here and if needed update the devops repo too

**Review by arvindh123 — CHANGES_REQUESTED** [review]
Please update

File: docs/kubernetes.md [hunk 2]
```diff
@@ -48,106 +48,153 @@ helm install ingress-nginx ingress-nginx/ingress-nginx --version 3.26.0 --create
...
+This will apply your changes to the existing installation. The following table lists the configurable parameters and their default values.
+
+| Parameter                          | Description                                                                                     | Default              |
+| ---------------------------------- | ----------------------------------------------------------------------------------------------- | -------------------- |
+| defaults.logLevel                  | Log level                                                                                       | info                 |
+| defaults.image.pullPolicy          | Docker Image Pull Policy                                                                        | IfNotPresent         |
+| defaults.image.rootRepository      | Docker Image Root Repository                                                                    | magistrala           |
+| defaults.image.repository          | Docker Image Repository                                                                         | magistrala           |
```
Comment (arvindh123) [inline docs/kubernetes.md:302]: I think this is removed , instead we have tag i guess, Please verify vaules.yaml in devops

**Review by arvindh123 — CHANGES_REQUESTED** [review]

File: docs/kubernetes.md (hunk 2 above)
Comment (JeffMboya) [reply docs/kubernetes.md:302]: > I think this is removed , instead we have tag i guess, Please verify vaules.yaml in devops

image

File: docs/kubernetes.md [hunk 3]
```diff
@@ -48,106 +48,153 @@ helm install ingress-nginx ingress-nginx/ingress-nginx --version 3.26.0 --create
...
+LAST DEPLOYED: Tue Aug 13 15:49:29 2024
+NAMESPACE: mg
+STATUS: deployed
+REVISION: 1
+```
+
+### Customizing a New Installation
 
-### Customizing Installation
+You can easily customize Magistrala during installtion by overriding the default settings using the `--set` option in Helm.
```
Comment (felixgateru) [inline docs/kubernetes.md]:
```suggestion
You can easily customize Magistrala during installation by overriding the default settings using the `--set` option in Helm.
```

File: docs/kubernetes.md [hunk 4]
```diff
@@ -167,15 +223,27 @@ List of add-ons services in charts:
...
+#### Step 1: Configure TCP and UDP Services
 
-Also, these three ports need to be exposed in the Service defined for the Ingress. You can do that with command that edit your service:
+The NGINX Ingress Controller uses ConfigMaps to expose TCP and UDP services. The necessary ConfigMaps are included in the Helm chart in the [ingress.yaml][ingress-yaml] file assuming that location of ConfigMaps should be `ingress-nginx/tcp-services` and `ingress-nginx/udp-services`. These locations are set with `--tcp-services-configmap` and `--udp-services-configmap` flags and you can check it in deployment of Ingress Controller or add it there in [args section for nginx-ingress-controller][ingress-controller-args] if it's not already specified. This is explained in [NGINX Ingress documentation][ingress-controller-tcp-udp]
+
+#### Step 2: Expose the Required Ports in the Ingress Service
+
+You need to expose the MQTT ports (`1883` for unencrypted and `8883` for encrypted messages) and the CoAP port (`5683` for UDP) in the NGINX Ingress Controller service. You can do that with the follwoing command that edits your service:
```
Comment (felixgateru) [inline docs/kubernetes.md]:
```suggestion
You need to expose the MQTT ports (`1883` for unencrypted and `8883` for encrypted messages) and the CoAP port (`5683` for UDP) in the NGINX Ingress Controller service. You can do that with the following command that edits your service:
```

**Review by felixgateru — CHANGES_REQUESTED** [review]

**Review by felixgateru — DISMISSED** [review]

File: docs/kubernetes.md [hunk 5]
```diff
@@ -48,106 +48,153 @@ helm install ingress-nginx ingress-nginx/ingress-nginx --version 3.26.0 --create
...
+| mqtt.proxy.mqttPort                | MQTT adapter proxy port                                                                         | 1884                 |
+| mqtt.proxy.wsPort                  | MQTT adapter proxy WS port                                                                      | 8081                 |
+| mqtt.broker.mqttPort               | MQTT adapter broker port                                                                        | 1883                 |
+| mqtt.broker.wsPort                 | MQTT adapter broker WS port                                                                     | 8080                 |
+| mqtt.broker.persistentVolume.size  | MQTT adapter broker data Persistent Volume size                                                 | 5Gi                  |
+| mqtt.redisESPort                   | MQTT adapter Event Store port                                                                   | 6379                 |
+| mqtt.redisCachePort                | MQTT adapter Redis Auth Cache port                                                              | 6379                 |
```
Comment (arvindh123) [inline docs/kubernetes.md:334]: @JeffMboya  Some of the values will not available in values.yaml, but template have values with defaults.

**Review by arvindh123 — CHANGES_REQUESTED** [review]

**Comment by dborovcanin** [comment]
@JeffMboya Is this up to date and ready?

**Comment by JeffMboya** [comment]
> @JeffMboya Is this up to date and ready?

@dborovcanin yes, this is up to date and ready

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**